### PR TITLE
Move the global usage into the IR function

### LINKED ISSUE #35: Explicit interface of entry points
SPIR-V lists all the inputs and outputs for each entry point. The question is whether it's a good idea to represent this somehow in the IR. See https://github.com/gpuweb/gpuweb/issues/592 for WGSL discussion.

Pros for specifying:
  - match SPIR-V
  - easier to translate from IR to anything, since we know the interface

Cons for specifying:
  - more verbose and redundant
  - incomplete - stuff like uniform or storage buffers are also technically the interface, so we still need to go and scan the uses of the globals

Another thought I had is that it doesn't make sense to specify that interface *per entry point*. It seems like it would make more sense to have for the function.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #35

### CODE REVIEW

File: src/back/msl.rs [hunk 1]
```diff
@@ -819,8 +813,10 @@ impl<W: Write> Writer<W> {
 
                 if em != spirv::ExecutionModel::GLCompute {
                     writeln!(self.out, "struct {} {{", location_input_name)?;
-                    for &handle in var_inputs.iter() {
-                        let var = &module.global_variables[handle];
+                    for ((handle, var), &usage) in module.global_variables.iter().zip(&fun.global_usage) {
```
Comment (Napokue) [inline src/back/msl.rs]: `module.global_variables.iter().zip(&fun.global_usage)` is used four times in this function. Isn't it possible to do this once and put it into a variable, to reduce iterations through `global_variables` and prevent to zip four times?

Comment (kvark) [reply src/back/msl.rs]: we could do it, yeah, although it's not going to save us any lines :)

Comment (Napokue) [reply src/back/msl.rs]: Less lines are not always better :) I was pointing it out more as a performance optimisation. I am not sure if there is a need for it though.

Comment (kvark) [reply src/back/msl.rs]: ah ok. There isn't any performance impact here. It's just an iteration over a few dozen of globals, super light.

Comment (Napokue) [reply src/back/msl.rs]: Sure, then it is all fine by me, rest looks good 👍

**Comment by kvark** [comment]
Appears that both @jrmuizel and @Napokue looked at this. Thank you!

**Pull request merged**